server/src/main/java/org/opensearch/index/store/remote/filecache/FileCachedIndexInput.java

```diff
@@ -133,8 +133,9 @@ public FileCachedIndexInput clone() {
 
     @Override
     public IndexInput slice(String sliceDescription, long offset, long length) throws IOException {
-        // never reach here!
-        throw new UnsupportedOperationException("FileCachedIndexInput couldn't be sliced.");
+        IndexInput slicedIndexInput = luceneIndexInput.slice(sliceDescription, offset, length);
+        cache.incRef(filePath);
+        return new FileCachedIndexInput(cache, filePath, slicedIndexInput, true);
```

Comment:
can we add test for this functionality to cover both full and block file case.?

Reply:
This IndexInput does not have any state whether we are calling it for a block/non-block file, so will behave in the same way for both block and non-block cases. So unit tests won't help here I guess.

Reply:
UT will be helpful to test that reference is getting incremented upon creating a slice, whether the offset of slice is correctly set or no based on underlying slice offset and length, then negative tests where slice is being created for invalid offset of length which is not part of underlying index input.